aospy_synthetic/test/test_db.py

```diff
@@ -3,78 +3,98 @@
-                                  intvl_in='monthly',
-                                  intvl_out='son',
-                                  dtype_in_time='ts',
-                                  dtype_in_vert='sigma',
-                                  dtype_out_time='avg',
-                                  dtype_out_vert=False,
-                                  level=False, backend=SQLAlchemyDB())
-        calc = Calc(test_calc)
-        calc.compute()
-        self.assertEqual(calc.intvl_in, 'monthly')
-
-    def test_region_calc(self):
-        test_calc = CalcInterface(proj=self.proj,
-                                  model=self.model,
-                                  run=self.run,
-                                  var=self.variable2,
-                                  date_range=('0021-01-01', '0080-12-31'),
-                                  intvl_in='monthly',
-                                  intvl_out='son',
-                                  dtype_in_time='ts',
-                                  dtype_in_vert='sigma',
-                                  dtype_out_time='avg',
-                                  dtype_out_vert=False,
-                                  region=regions.nh,
-                                  level=False, backend=SQLAlchemyDB())
-        calc = Calc(test_calc)
-        calc.compute()
-        self.assertEqual(calc.intvl_in, 'monthly')
+        call(['rm', 'test.db'])
+
+    def test_add(self):
+        self.db.add(self.AospyObj)
+        with self.db.session_scope() as session:
+            q = session.query(self.db_cls)
+            db_obj = q.filter_by(hash=hash(self.AospyObj)).first()
+
+            for attr in self.db_cls._metadata_attrs:
+                actual = getattr(db_obj, attr)
+                expected = getattr(self.AospyObj,
+                                   self.db_cls._metadata_attrs[attr])
+                self.assertEqual(actual, expected)
+
+    def test_uniqueness_checking(self):
+        self.db.add(self.AospyObj)
+        self.db.add(self.AospyObj)  # Call add on a duplicate object
+        with self.db.session_scope() as session:
+            q = session.query(self.db_cls)
+            db_objs = q.filter_by(hash=hash(self.AospyObj)).all()
+            self.assertEqual(len(db_objs), 1)
+
+    @unittest.expectedFailure
+    def test_update_attr(self):
+        self.db.add(self.AospyObj)
+        setattr(self.AospyObj, self.ex_str_attr, 'updated')
+        self.db.add(self.AospyObj)
+        with self.db.session_scope() as session:
+            q = session.query(self.db_cls)
+            db_objs = q.filter_by(hash=hash(self.AospyObj)).all()
+            self.assertEqual(len(db_objs), 1)  # No duplicates
+
+            db_obj = db_objs[0]
+            actual = getattr(db_obj, self.ex_str_attr)
+            expected = 'updated'
+            self.assertEqual(actual, expected)
+
+
+class TestModelDB(TestProjDB):
```

Comment:
I will take a look at what you've done in ```infinite-diff```;  making shared superclass as you suggest sounds like the cleanest way to go.